src/main/java/Duke.java

```diff
@@ -1,10 +1,272 @@
+                    String deadlineDate = userInput.substring(deadlineDateIndex + 3, userInput.length());
+                    Task newTask = addDeadline(title.trim(), deadlineDate.trim());
+                    printSuccessfulAddMessage(newTask);
+                } catch (StringIndexOutOfBoundsException e) {
+                    separator();
+                    System.out.println(EmptyDeadlineException.errorMessage());
+                    separator();
+                }
+            } else if (command.equals("event")) {
+                try {
+                    int atIndex = userInput.indexOf("/at");
+                    String title = userInput.substring(0, atIndex).replace("event", "");
+                    String atDate = userInput.substring(atIndex+3, userInput.length());
+                    Task newTask = addEvent(title.trim(), atDate.trim());
+                    printSuccessfulAddMessage(newTask);
+                } catch (StringIndexOutOfBoundsException e) {
+                    separator();
+                    System.out.println(EmptyEventException.errorMessage());
+                    separator();
+                }
+            } else if (command.equals("delete")) {
+                try {
+                    delete(Integer.parseInt(userInputArray[1]));
+                } catch (IndexOutOfBoundsException e) {
+                    separator();
+                    System.out.println("☹ OOPS!!! A task with that ID does not exist.");
+                    separator();
+                }
+            } else {
+                try {
+                    throw new InvalidCommandException();
+                } catch (InvalidCommandException e) {
+                    separator();
+                    System.out.println(InvalidCommandException.errorMessage());
+                    separator();
+                }
+            }
+
+        }
+        save();
+        exit();
     }
-}
+
+    public static void delete(int taskId) {
+        Task task = items.get(taskId - 1);
+        items.remove(task);
+        separator();
+        System.out.println("Noted. I've removed this task: ");
+        System.out.println(" " + task.getTitle());
+        System.out.println("Now you have " + items.size() + " tasks in the list.");
+        separator();
+    }
+
+    public static void save() {
+        try {
+            // empty current saved items;
+            FileWriter clear = new FileWriter(f);
+            clear.write("");
+            clear.close();
+            int i = 0;
+            for(i = 0; i < Task.numberOfTasks; i++) {
+                Task task = items.get(i);
+
+                FileWriter fw = new FileWriter(f, true);
+
+                fw.write(task.saveFormat() + "\n");
+                fw.close();
+
+            }
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+
+    }
+
+    public static void initData() {
```

Comment:
Again, you can improve the SLAP in this method by extracting out lower-level operations to another method.